src/librustdoc/html/static/js/main.js

```diff
@@ -587,6 +588,177 @@ function preLoadCss(cssUrl) {
         window.register_implementors(window.pending_implementors);
     }
 
+    // <https://github.com/search?q=repo%3Arust-lang%2Frust+[RUSTDOCIMPL]+type.impl&type=code>
+    window.register_type_impls = imp => {
```

Comment:
Also, could you add some comments explaining a bit more what it is manipulating and what it's generating? Not sure what the best format would be here though. There are a lot of smaller "sections" in the code, so maybe have a comment for each section explaining what it's doing instead of a big one explaining everything at the start? What do you think?